src/boards/rainbow_esp.cpp

```diff
@@ -1020,32 +1053,71 @@ void BrokeStudioFirmware::httpdEvent(mg_connection *nc, int ev, void *ev_data) {
 			}
 
 			// Process request
-			std::map<std::string, std::string>::const_iterator file_data = params.find("file");
-			std::map<std::string, std::string>::const_iterator filename = params.find("path");
-			if (file_data == params.end() || filename == params.end()) {
-				send_generic_error();
-				return;
-			}
-			std::pair<uint8, uint8> path = BrokeStudioFirmware::pathIndexFromStr(filename->second);
-			if (path.second == NO_WORKING_FILE) {
+			std::map<std::string, std::string>::const_iterator file_data = params.find("file_data");
+			std::map<std::string, std::string>::const_iterator path = params.find("path");
+			std::map<std::string, std::string>::const_iterator file = params.find("file");
+
+			if (file_data == params.end() || path == params.end() || file == params.end()) {
 				send_generic_error();
 				return;
 			}
-			self->files[path.first][path.second] = std::vector<uint8>(file_data->second.begin(), file_data->second.end());
-			self->file_exists[path.first][path.second] = true;
+
+			uint8 path_index = std::atoi(path->second.c_str());
+			uint8 file_index = std::atoi(file->second.c_str());
+
+			self->files[path_index][file_index] = std::vector<uint8>(file_data->second.begin(), file_data->second.end());
+			self->file_exists[path_index][file_index] = true;
 
 			UDBG("Upload sucessful\n");
 
 			// Return something webbrowser friendly
-			send_message(200, "<html><body><p>Upload success</p></body></html>\n", "text/html");
+			send_message(200, "{\"success\":\"true\"}\n", "application/json");
 		}else {
 			char const* www_root = ::getenv("RAINBOW_WWW_ROOT");
 			if (www_root == NULL) {
-				send_message(
-					200,
-					R"-(<html><body><form action="/api/file/upload" method="post" enctype="multipart/form-data"><input name="file" type="file"><br /><input name="path" type="text" value="/USER/file10.bin"><br /><button type="submit">Upload</button></form></body></html>)-",
-					"text/html"
-				);
+				std::string upload_form = R"-(<!doctype html><html><head><style>*{margin:2px}body{font-family:Arial}</style></style>)-";
+				upload_form += R"-(<body><h1>Upload file to Rainbow:</h1>)-";
+				// Let's comment and keep the classic/simple HTML stuff for now, just in case
+				//upload_form += R"-(<form action="/api/file/upload" method="post" enctype="multipart/form-data">)-";
```

Comment:
Not fan of letting such comment that basicaly says "we are not doing that". Just remove it and let the reader focus on what is actually implemented.